src/XR/features/WebXREyeTracking.ts

```diff
@@ -0,0 +1,151 @@
+import { Ray } from "../../Culling/ray";
+import { Nullable } from "../../types";
+
+/**
+ * The WebXR Eye Tracking feature grabs eye data from the device and provides it in an easy-access format.
+ * Currently only enabled for BabylonNative applications, developers intending to use this must enable the Gaze Input capability in their appx package manifest for data to flow.
+ */
+export class WebXREyeTracking extends WebXRAbstractFeature {
+    private _latestEyeSpace: Nullable<XRSpace>;
+    private _gazeRay: Nullable<Ray>;
+
+    private _tmpQuat: Quaternion = new Quaternion();
+    private _eventListeners: { [event in XREventType]?: (event: XREyeTrackingSourceEvent) => void };
+
+    /**
+     * The module's name
+     */
+    public static readonly Name = WebXRFeatureName.EYE_TRACKING;
+    /**
+     * The (Babylon) version of this module.
+     * This is an integer representing the implementation version.
+     * This number does not correspond to the WebXR specs version
+     */
+    public static readonly Version = 1;
+
+    /**
+     * This observable will notify registered observers when eye tracking starts
+     */
+    public onEyeTrackingStartedObservable: Observable<Ray> = new Observable();
+    /**
+     * This observable will notify registered observers when eye tracking ends
+     */
+    public onEyeTrackingEndedObservable: Observable<void> = new Observable();
+    /**
+     * This observable will notify registered observers on each frame that has valid tracking
+     */
+    public onEyeTrackingFrameUpdateObservable: Observable<Ray> = new Observable();
+
+    /**
+     * Creates a new instance of the XR eye tracking feature.
+     * @param _xrSessionManager An instance of WebXRSessionManager.
+     */
+    constructor(_xrSessionManager: WebXRSessionManager) {
+        super(_xrSessionManager);
+        this.xrNativeFeatureName = "eye-tracking";
+        if (this._xrSessionManager.session) {
+            this._init();
+        } else {
+            this._xrSessionManager.onXRSessionInit.addOnce(() => {
+                this._init();
+            });
+        }
+    }
+
+    /**
+     * Dispose this feature and all of the resources attached.
+     */
+    public dispose(): void {
+        super.dispose();
+
+        Object.keys(this._eventListeners).forEach((eventName: string) => {
+            const func = this._eventListeners && this._eventListeners[eventName as XREventType];
+            if (func) {
+                this._xrSessionManager.session.removeEventListener(eventName as XREventType, func);
+            }
+        });
+
+        this.onEyeTrackingStartedObservable.clear();
+        this.onEyeTrackingEndedObservable.clear();
+        this.onEyeTrackingFrameUpdateObservable.clear();
+    }
+
+    /**
+     * Returns whether the gaze data is valid or not
+     * @returns true if the data is valid
+     */
+    public isEyeGazeValid(): boolean {
+        return !!this._gazeRay;
+    }

```

Comment:
Should this be a property?
```suggestion
    public get isEyeGazeValid(): boolean {
        return !!this._gazeRay;
    }

```